app/models/bank_account.rb

```diff
@@ -65,12 +65,24 @@ def primary_payment_provider
       raise "Bankable #{bankable.inspect} doesn't respond to :primary_payment_provider"
     end
   end
+
   private
 
   def account_is_unique_to_bankable
-    accounts = bankable.bank_accounts.visible.where(account_type: account_type, last_four: last_four, bank_name: bank_name, name: name)
+    unique_params = {
+      account_type: account_type,
+      last_four: last_four,
+      bank_name: bank_name,
+      name: name
+    }
+
+    if credit_card?
+      unique_params.merge!(expiration_month: expiration_month, expiration_year: expiration_year)
+    end
```

Comment:
@thermistor not so happy with this `if` here as we should likely split out bank_account into two subclasses (one for checking, one for credit card? Or, not sure if better Ruby way. Anyway, I think it's okay for now, but await your thoughts.

Reply:
It's ok for now.